### PR TITLE
fix: Generate APP_STATE on every login request

### PR DESCRIPTION
**Pull request by bernardo-bastos**
Generates a random string for `state` field on the `/login` request

### CODE REVIEW

File: pkg/auth/dex.go [hunk 1]
```diff
@@ -182,8 +193,13 @@ func (a *DexAppClient) handleDexLogin(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// TODO(BB) Set an app state to make the connection more secure
-	authCodeURL := oauthConfig.AuthCodeURL("APP_STATE")
+	// TODO(BB) Set an app state to make the connection more secure.
+	// Currently a random string is generated but a session state should be built.
+	state, err := generateState()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (sven-urbanski-freiheit-com) [inline pkg/auth/dex.go:200]:
```suggestion
		http.Error(w, fmt.Errorf("could not generate state %s", err.Error()), http.StatusInternalServerError)
```

**Review by sven-urbanski-freiheit-com — CHANGES_REQUESTED** [review]
.

File: pkg/auth/dex.go [hunk 2]
```diff
@@ -182,8 +193,13 @@ func (a *DexAppClient) handleDexLogin(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// TODO(BB) Set an app state to make the connection more secure
-	authCodeURL := oauthConfig.AuthCodeURL("APP_STATE")
+	// TODO(BB) Set an app state to make the connection more secure.
```
[the remark this answers is not in the document]
Comment (bernardo-bastos) [reply pkg/auth/dex.go]: Removed the comment!

File: pkg/auth/dex.go (hunk 1 above)
Comment (bernardo-bastos) [reply pkg/auth/dex.go]: Replaced it with a better error message.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**